xgboost_ray/main.py

```diff
@@ -436,6 +509,51 @@ def _autodetect_resources(ray_params: Union[None, RayParams, Dict] = None,
     return cpus_per_actor, gpus_per_actor
 
 
+def _num_possible_actors(num_cpus_per_actor: int,
+                         num_gpus_per_actor: int,
+                         resources_per_actor: Optional[Dict] = None,
+                         max_needed: int = -1) -> int:
+    """Returns number of actors that could be scheduled on this cluster."""
+    if max_needed < 0:
```

Comment:
Can we move this + helper functions to a util file? Would it make sense to do that?

Reply:
Yep, I thought about this, too. Moved it to `elastic.py` (and some things to `util.py`) - we'll need local imports now though to avoid circular imports